util/gctuner/tuner.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gctuner
+
+import (
+	"math"
+	"os"
+	"strconv"
+	"sync/atomic"
+
+	"github.com/pingcap/tidb/util"
+)
+
+const (
+	// MaxGCPercent is the default max cost of memory.
+	MaxGCPercent uint32 = 500
+	// MinGCPercent is the default min cost of memory.
+	MinGCPercent uint32 = 20
+)
+
+var defaultGCPercent uint32 = 100
```

Comment:
a good suggestion. let me improve it.

Reply:
Done

Reply:
could we need a interface to let user could change threshod? for example set a sysvar to change gc threshold to make gc more controlable?

Reply:
Yes, It will have. this PR is a simple implementation. it still needs to do many improvements. so Now it has a switch to control whether to enable ```gogctuner```.

Reply:
Now we have a switch to control whether to enable the ```gogctuner```. But after merging this PR, we will add the HTTP API for a user to control the threshold. If it adds to this PR. This pr will be too big to be reviewed.